plugins/uefi-mok/fu-uefi-mok-plugin.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-uefi-mok-common.h"
+#include "fu-uefi-mok-plugin.h"
+
+struct _FuUefiMokPlugin {
+	FuPlugin parent_instance;
+};
+
+G_DEFINE_TYPE(FuUefiMokPlugin, fu_uefi_mok_plugin, FU_TYPE_PLUGIN)
+
+static gchar *
+fu_uefi_mok_plugin_get_filename(void)
+{
+	g_autofree gchar *sysfsdir = fu_path_from_kind(FU_PATH_KIND_SYSFSDIR_FW);
+	return g_build_filename(sysfsdir, "efi", "mok-variables", "HSIStatus", NULL);
+}
+
+static gboolean
+fu_uefi_mok_plugin_startup(FuPlugin *plugin, FuProgress *progress, GError **error)
+{
+	g_autofree gchar *fn = fu_uefi_mok_plugin_get_filename();
+
+	/* sanity check */
+	if (!g_file_test(fn, G_FILE_TEST_EXISTS)) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED, "%s does not exist", fn);
+		return FALSE;
+	}
```

Comment:
this basically means that there is too old of a kernel, right?  Do we want to still tell them to upgrade kernels so they can get this information?

Reply:
Got it.  No concern then.